Doc/library/dataclasses.rst

```diff
@@ -355,6 +355,12 @@ Module contents
 
      assert astuple(p) == (10, 20)
      assert astuple(c) == ([(0, 0), (10, 4)],)
+     
+   The tuple is a :func:`copy.deepcopy` of contained objects, which may
+   be an issue in some corner cases. To create a shallow copy, the
+   following workaround may be used::
```

Comment:
I think this should mention that it recurses into dataclasses, namedtuples, lists, tuples, and dicts, and deep copies anything else.

Reply:
While it does recurse into lists etc, it also performs, in effect, a "deepcopy" on them, in the sense that contained lists, dicts, etc, will not be the same objects as in the original dataclass -- that's what I was trying to emphasize here.

So it's a bit tricky to combine the previous paragraph and this one.

If the reader reads the previous paragraph and then this one, they should have a good understanding of what happens, but if they skip first and just read this one, they will be mislead.

Would it help if instead of saying with the link :func:`copy.deepcopy`, I would  say "a 'deepcopy' process is performed on the contained objects, hinting that it's analogous to deepcopy but not literally running deepcopy on it?

Another option might be to move the last 2 sentences from the previous paragraph into this paragraph (sentences starting with 'Each dataclass....'), similar to what you suggested. The downside is that it will be harder to read and get to the more important point that dataclasses contained in lists etc will all be converted to tuples. This paragraph covers a much more of an edge case that won't interest most people..

Reply:
Then I would just add a sentence to the previous paragraph, saying "Other objects are copied with copy.deepcopy".

Then have the last paragraph mention the workaround, without mentioning deepcopy again.